File: online-training-platform/leetcode/valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        
        // Rows
        for(int r = 0 ; r < 9 ; r++) {
            set<char> numbers;
            for(int c = 0 ; c < 9 ; c++) {
                
                if(board[r][c] == '.'){
                    continue;
                }
                
                if(numbers.find(board[r][c]) != numbers.end()) {
                    return false;
                }
                numbers.insert(board[r][c]);
            }
        }
        
        // Colum
        for(int c = 0 ; c < 9 ; c++) {
            set<char> numbers;
            for(int r = 0 ; r < 9 ; r++) {
                
                if(board[r][c] == '.'){
                    continue;
                }
                
                if(numbers.find(board[r][c]) != numbers.end()) {
                    return false;
                }
                numbers.insert(board[r][c]);
            }
        }
        
        // Square
        for(int c = 0 ; c < 9 ; c += 3) {
            for(int r = 0 ; r < 9 ; r += 3) {
                
                set<char> numbers;
                for(int i = r ; i < r + 3 ; i++) {
                    for(int j = c ; j < c + 3 ; j++) {
                        if(board[i][j] == '.'){
                            continue;
                        }

                        if(numbers.find(board[i][j]) != numbers.end()) {
                            return false;
                        }
                        numbers.insert(board[i][j]);        
                    }
                }   
            }
        }
        return true;
    }
};

File: online-training-platform/leetcode/valid-sudoku.kt
class Solution {
    fun isValidSudoku(board: Array<CharArray>): Boolean {
        
        // Row
        val row = HashSet<Char>()
        for(r in 0 until 9){
            row.clear()
            for(c in 0 until 9){
                if(board[r][c] == '.') continue
                if( row.contains(board[r][c])){
                    return false
                }
                row.add(board[r][c])
            }
        }
        
        // Colum
        val colum = HashSet<Char>()
        for(c in 0 until 9){
            colum.clear()
            for(r in 0 until 9){
                if(board[r][c] == '.') continue
                if(colum.contains(board[r][c])){
                    return false
                }
                colum.add(board[r][c])
            }
        }
        
        
        // Square
        val box = HashSet<Char>()
        for(r in 0 until 9 step 3){
            for(c in 0 until 9 step 3){
                box.clear()
                for(i in r until r + 3){
                    for(j in c until c + 3){
                        if(board[i][j] == '.') continue
                        if(box.contains(board[i][j])){
                            return false
                        }
                        box.add(board[i][j])
                    }
                }
            }
        }
        
        return true
    }
}